### PR TITLE
Feat: add new OIDC credentials creation and assigment (Azure data sou…

### PR DESCRIPTION
**Pull request by TomerHeber**
…rce)

Part of #743 

### Solution

1. Refactored to use common code.
2. Moved oidc jwt key to organization data resource (makes more sense).
3. Added azure oidc data source.
4. Added integration tests.
5. Added acceptance tests for azure oidc data source.

### CODE REVIEW

File: env0/data_aws_oidc_credentials_test.go [hunk 1]
```diff
@@ -29,8 +29,6 @@ func TestAwsOidcCredentialDataSource(t *testing.T) {
 		Type: string(client.AwsAssumedRoleCredentialsType),
 	}
 
-	oidcSub := "oidc sub 123345 !!!"
```
Comment (TomerHeber) [inline env0/data_aws_oidc_credentials_test.go:32]: moved to organization data.

File: env0/data_azure_oidc_credentials_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/env0/terraform-provider-env0/client/http"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAzureOidcCredentialDataSource(t *testing.T) {
```
Comment (TomerHeber) [inline env0/data_azure_oidc_credentials_test.go:13]: tests duplication to be refactored on the next commit.

File: env0/data_organization.go [hunk 3]
```diff
@@ -49,5 +54,10 @@ func dataOrganizationRead(ctx context.Context, d *schema.ResourceData, meta inte
 		return diag.Errorf("schema resource data serialization failed: %v", err)
 	}
 
+	oidcSub, err := apiClient.OidcSub()
+	if err == nil {
```
Comment (TomerHeber) [inline env0/data_organization.go:58]: may fail if oidc is disabled. Set the value if successful.

File: env0/data_oidc_credentials.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+func dataOidcCredentialRead(credentialsType CloudType) func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+		var credentials client.Credentials
+		var err error
+
+		id, ok := d.GetOk("id")
+		if ok {
+			credentials, err = getCredentialsById(id.(string), credentialsTypeToPrefixList[credentialsType], meta)
+		} else {
+			credentials, err = getCredentialsByName(d.Get("name").(string), credentialsTypeToPrefixList[credentialsType], meta)
```
Comment (Yossi-kerner) [inline env0/data_oidc_credentials.go:42]: don't we want to check that `d.Get("name")` is ok?

**Review by Yossi-kerner — APPROVED** [review]
Looks good left one question

File: env0/data_oidc_credentials.go (hunk 4 above)
Comment (TomerHeber) [reply env0/data_oidc_credentials.go:42]: not required. This is enforced in the schema validation level ("exactlyOneOf").

**Pull request merged**